File: mobile/app/src/main/java/com/sobczal2/biteright/dto/products/ProductSortingStrategy.kt
package com.sobczal2.biteright.dto.products

enum class ProductSortingStrategy(val value: Int) {
    NameAsc(0),
    NameDesc(1),
    ExpirationDateAsc(2),
    ExpirationDateDesc(3),
    AddedDateTimeAsc(4),
    AddedDateTimeDesc(5),
    ConsumptionAsc(6),
    ConsumptionDesc(7);

    companion object {
        fun fromInt(value: Int) = entries.firstOrNull { it.value == value }
    }
}

File: backend/src/BiteRight.Application/Dtos/Products/ProductSortingStrategy.cs
// # ==============================================================================
// # Solution: BiteRight
// # File: ProductSortingStrategy.cs
// # Author: Łukasz Sobczak
// # Created: 12-02-2024
// # ==============================================================================

namespace BiteRight.Application.Dtos.Products;

// TODO na ekranie z listą wszystkich produktów,
// sortowanie, wyszukiwanie(nazwa, kategoria, opisie)
// i filtrowanie(po stanie daty ważnośc - expired, nie expired,
// po kategorii)
public enum ProductSortingStrategy
{
    NameAsc = 0,
    NameDesc = 1,
    ExpirationDateAsc = 2,
    ExpirationDateDesc = 3,
    AddedDateTimeAsc = 4,
    AddedDateTimeDesc = 5,
    PercentageAmountAsc = 6,
    PercentageAmountDesc = 7,
    Default = NameAsc
}